sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -872,16 +867,18 @@ def download_manifest(self, repository, tag_or_digest, **kwargs):
                     **kwargs
                 )
             )
-            digest = response.http_response.headers['Docker-Content-Digest']
             manifest = OCIManifest.deserialize(cast(ManifestWrapper, manifest_wrapper).serialize())
             manifest_stream = _serialize_manifest(manifest)
+            if tag_or_digest.startswith("sha:256:"):
+                digest = tag_or_digest
+            else:
+                digest = response.http_response.headers['Docker-Content-Digest']
```

Comment:
This logic seems a bit strange?
Why aren't we consistent about whether we check against the service provided value or the client supplied value? Is this behaviour documented/expected by customers?

Reply:
When the manifest is requested via `tag` rather than `digest`, there is no pre-request digest value to use to validate the digest returned from the client. We instead compare the digest in the response header to the digest computed by the payload.

Reply:
@lmolkova is this the approach across languages?
@YalinLi0312 is this is expected, then this behaviour needs to be documented in the `tag_or_digest` parameter description.

Reply:
Yes, it is confirmed across languages.

Reply:
> @lmolkova is this the approach across languages? @YalinLi0312 is this is expected, then this behaviour needs to be documented in the `tag_or_digest` parameter description.

@annatisch yep, we're checking if data was not corrupted or tampered with. The service team asked us to do it and we checked that docker cli does the same. We should definitely document it though.

Reply:
Thanks for confirmation :)